msal/application.py

```diff
@@ -383,17 +387,20 @@ def acquire_token_by_username_password(
 
     def _acquire_token_by_username_password_federated(
             self, user_realm_result, username, password, scopes=None, **kwargs):
+        verify = kwargs.pop("verify", self.verify)
+        proxies = kwargs.pop("proxies", self.proxies)
```

Comment:
Guess we must have reasons for taking related settings through arguments, but I do feel it is a bit redundant. So wonder why?

Reply:
Excellent question! The use of `**kwargs` a.k.a. Keyworded Variables is not a requirement in terms of authentication api design, that's why they are [purposely undocumented](https://msal-python.readthedocs.io/en/refine-docs/). The behavior remains here, only as an influence by the underlying requests, which can accept [verify](http://docs.python-requests.org/en/master/user/advanced/#ssl-cert-verification), [proxies](http://docs.python-requests.org/en/master/user/advanced/#proxies), and some more [other parameters](http://docs.python-requests.org/en/master/user/quickstart/#timeouts), all through ad-hoc arguments. And we get these [syntactic sugar](https://en.wikipedia.org/wiki/Syntactic_sugar), almost for free, by wiring them properly.

All that being said, I would personally have no objection if many of you prefer to remove all those `**kwargs` from MSAL Python APIs. Just let us know.

Reply:
I suggest don't remove **kwargs, and also don''t extract `verify` and `proxies` out of the kwargs. My main point is we should only have one place to set these 2 fields which is through the constructor

Reply:
Understood. But:

1. I think our current approach, i.e. only documenting `verify` and `proxies` in constructor, is conceptually close enough to "we only have one place to set these 2 fields which is through the constructor".
2. As long as we keep the **kwargs in this method signature, it is possible that the caller still provides `verify` and `proxies`. Then what shall we do in this undocumented scenario? The current implementation tries to provide an arguably intuitive behavior matching the caller's intention. Rather than, accepting the `verify` and `proxies` yet silently ignore them, even though it would technically be the "correct" behavior for any undocumented/unimplemented parameter.
3. If, however, we really don't like the undocumented behavior in the previous paragraph, we could choose go one step further by removing the **kwargs in this method signature. That way it would become impossible to provide `verify` and `proxies` through this method.

Reply:
Okay, got it. For context, I was a bit confused by why we only inspect the kwargs here, while `acquire_token_silent` above doesn't have the inspections

Reply:
Yes, that seemingly discrepancy was necessary, only for this method, because the `mex_send_request(...)` and `wstrust_send_request(...)` happen to require an explicit `verify` and `proxies`. All in all, personally I think the current approach is good enough.
Guess we can conclude this conversation thread, then?